src/msg/async/ProtocolV2.cc

```diff
@@ -2790,6 +2790,11 @@ CtPtr ProtocolV2::reuse_connection(const AsyncConnectionRef& existing,
           existing->open_write = false;
           exproto->session_stream_handlers = std::move(temp_stream_handlers);
           exproto->session_compression_handlers = std::move(temp_compression_handlers);
+          if (!reconnecting) {
+            exproto->peer_supported_features = peer_supported_features;
```

Comment:
I won't pretend to understand the details of this `reuse_connection` business --  it always appeared super fragile to me -- but the fix looks sensible and in line with `write_message` being called with no locks whatsoever.

One nit: does `peer_supported_features` really need this in-event-center-thread treatment?  I think it could be left alone because if assigning `peer_supported_features` there is wrong, assigning `connection_features` is likely wrong too.

Reply:
Also, the title of the commit (and the PR) is misleading.  `set_is_rev1` is under `write_lock` already but that isn't helping because `write_message` doesn't hold any locks when calling `append_frame`.  The fix is really about moving the reset into the right thread, isn't it?

Reply:
Yes, you're right! When writing the commit title I assumed that holding the `write_lock` solves the problem. When i retake a look on my own comments in the tracker, I see this isn't the case. Fixing!

Reply:
> One nit: does `peer_supported_features` really need this in-event-center-thread treatment?  I think it could be left alone because if assigning `peer_supported_features` there is wrong, assigning `connection_features` is likely wrong too.

It rather unneeded. I think the sole reason behind moving `peer_supported_features` was to keep similar things together.

Reply:
> I think the sole reason behind moving peer_supported_features was to keep similar things together.

In that sense `peer_supported_features` is much more similar to `connection_features` than to `rx/tx_frame_asm`.

Reply:
OK, this as a good analogy. Moved.